docs/read-more/how-to-write-a-sequence.md

```diff
@@ -0,0 +1,251 @@
+
+    return out
+}
+```
+
+### Typescript
+Sequences that transform data should be typed as [TransformApp](https://hub.scramjet.org/docs/types/modules#transformapp).  
+[Here's an example](https://github.com/scramjetorg/scramjet-cloud-docs/blob/main/samples/transform-string-stream/src/index.ts).
+
+## Sequence arguments
+Every sequence can be spawned with arbitrary number of arguments
+```bash
+si seq start <sequence-id> --args ['Hello', 123, { abc: 456 }]
+```
+you can access this args using function parameters in your sequence:
+```ts
+export default function(input, param1, param2., param3) {
+    console.log(param1 + ' ' param2 + ' ' param3.abc)
+    // Prints "Hello 123 456" to stdout
+}
+```
+
+## Sending data between sequences (topics)
+Sometimes you need a bunch of sequences to talk to each other. Topics are the solution. It’s a PubSub system that allows for many writing and reading instances to exchange data. 
+Topics are either attached to the input stream or routed from the output stream.
+### Writing to a topic 
+To have your output stream routed to a topic, it needs to have a property `topic` with a topic name and `contentType` for proper consumer encoding. 
+```ts
+export default function() {
+    const out = new PassThrough()
+
+    temparatureSensor.on('update', (data) => {
+        out.write(data)
+    })
+
+    out.topic = 'temperature-readouts'
+    out.contentType = 'text/plain'
+
+    return out
+}
+```
+### Reading from a topic 
+To send topic as input stream we specify a topic config object with two properites: `requires` - with a name of topic, and `contentType` for ensuring proper encoding.
+```ts
+const CRITICAL_TEMP_CELCIUS = 40
+
+const app = [
+    {
+        requires: 'temperature-readouts',
+        contentType: 'text/plain'
+    },
+    async function* (input) {
+        for await (const data of input) {
+            if(+data > CRITICAL_TEMP_CELCIUS) {
+                this.logger.trace('Temparature exceeded critical level')
+            }
+        } 
+    }
+]
+
+export default app
+```
+### Typescript
+Writing to a topic requires specifying two additional properties that might not be present on your output stream. You can extend the type of your output stream by [HasTopicInformation](https://hub.scramjet.org/docs/types/modules#hastopicinformation) type.  
+Reading from a topic would require you to type your app as a tuple similar to this example:
+```ts
+const app: [{requires: string, contentType: string}, ReadableApp] = [
+    { requires: 'hello', contentType: 'text/plain' },
+    function(input) { ... }
+]
+```
+
+### Interacting with topics
+Apart from sequences communicating between each other you can also feed/consume a topic using our CLI, REST API, or API Client. 
+
+## Standard streams (stdin/stdout/stderr)
+Every sequence has access to standard streams of a program. You can read data from stdin. Send additional information to stdout and to stderr. These are separate from input/output streams.
```

Comment:
Perhaps it's obvious for some but I'd add that e.g. console.log(..) goes to /stdout because I've seen this confusion especially when /log enters the game :)

Reply:
fixed